lib/test-service-provider.js

```diff
@@ -46,42 +46,58 @@
 	 */
 	Box.TestServiceProvider = function(serviceStubs, allowedServicesList) {
 		this.stubs = serviceStubs || {};
-		this.allowedServices = {};
-
-		// Make lookup fast, convert to a map
-		var allowedServices = allowedServicesList || [];
-		for (var i = 0, len = allowedServices.length; i < len; i++) {
-			this.allowedServices[allowedServices[i]] = true;
-		}
+		this.allowedServicesList = allowedServicesList || [];
+		this.serviceInstances = {}; // Stores the instances of pre-registered services
 	};
 
 	Box.TestServiceProvider.prototype = {
 
 		/**
+		 * Check if service is allowed. This should be replaced by Array.prototype.indexOf when we drop IE 8 support
+		 * @param {string} serviceName The name of the service being checked
+		 * @returns {boolean} Returns true if service is in the allowed list
+		 */
+		isServiceAllowed: function(serviceName) {
```

Comment:
I originally made it private but that meant passing in the this.allowedServiceList as a parameter so I opted for just another interface method (since TestServiceProvider doesn't really match up a context/application object anyway). Not sure, think it's worth making private?